test/backend/db_tests.js

```diff
@@ -450,6 +451,49 @@ module.exports = function(config, DB) {
         )
 
         test(
+          'locked accounts',
+          function (t) {
+            t.plan(5)
+
+            var lockedAt = Date.now()
+            var unlockCode = hex16()
+            var uid = Buffer(ACCOUNT.uid)
+            var email = Buffer(ACCOUNT.email)
+
+              // set lockedAt
+            return db.lockAccount(uid, { lockedAt: lockedAt, unlockCode: unlockCode })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .lockAccount() request' + String(err))
+              })
+              .then(function(result) {
+                t.deepEqual(result, {}, 'Returned an empty object for lockAccount')
+                return db.account(uid)
+              })
+              .then(function(account) {
+                t.equal(account.lockedAt, lockedAt, 'account should now be locked')
+                return db.unlockCode(uid)
+              })
+              .then(function (_unlockCode) {
+                t.deepEqual(_unlockCode.unlockCode, unlockCode, 'unlockCode should be set')
+
+                // try to unlock the account
+                return db.unlockAccount(uid)
+              })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .unlockAccount() request' + String(err));
+              })
+              .then(function(result) {
+                t.deepEqual(result, {}, 'Returned an empty object for unlockAccount')
+                // now check it's been saved
+                return db.emailRecord(email)
```

Comment:
In many other tests where the accounts table has changed, I try and then test both `.account()` and also `.emailRecord()` at the same time to make sure they return what we expect. Therefore here it'd be nice if there was an `.emailRecord()` call above (to make sure the `lockedAt` is set and a `.account()` call here to also check it is now unset. Does that make sense? It's super specific but I wonder if it'll be more complete.